### PR TITLE
[CompilerPerf] Added benchmark comparisons of older versions of FSharp.Compiler.Service

### PR DESCRIPTION
**Pull request by TIHan**
This adds benchmark comparisons for older versions of FSharp.Compiler.Service to what is currently in the branch. This includes the notebook to run and display the results.

Here is an example of the results:
newplot

### CODE REVIEW

**Comment by TIHan** [comment]
This is ready @KevinRansom

File: tests/benchmarks/BenchmarkComparison/runner.ipynb [hunk 1]
```diff
@@ -0,0 +1,234 @@
...
+   "source": [
+    "let benchmarkData =\r\n",
+    "    [\r\n",
+    "        // Note: SERVICE_30_0_0 and SERVICE_13_0_0 refer to define constants in order to build the benchmark on older versions.\r\n",
+    "      //  benchmarkCurrent()\r\n",
+    "      benchmarkCommit \"81d1d918740e9ba3cb2eb063b6f28c3139ca9cfa\" \"\"\r\n",
+    "      benchmarkCommit \"1d36c75225436f8a7d30c4691f20d6118b657fec\" \"\"\r\n",
+    "      benchmarkCommit \"2e4096153972abedae142da85cac2ffbcf57fe0a\" \"\"\r\n",
+    "      benchmarkCommit \"af6ff33b5bc15951a6854bdf3b226db8f0e28b56\" \"\"\r\n",
+    "      //  benchmarkVersion \"40.0.0 (6/22/2021)\" \"40.0.0\" \"\"\r\n",
```
Comment (KevinRansom) [inline tests/benchmarks/BenchmarkComparison/runner.ipynb:133]: Commments

**Review by KevinRansom — APPROVED** [review]

File: tests/benchmarks/BenchmarkComparison/Program.fs [hunk 2]
```diff
@@ -0,0 +1,11 @@
+﻿
+open System.IO
+open System.Reflection
+open BenchmarkDotNet.Running
+open BenchmarkComparison
+
+[<EntryPoint>]
+let main _ =
+    BenchmarkRunner.Run<TypeCheckingBenchmark1>() |> ignore
```
Comment (vzarytovskii) [inline tests/benchmarks/BenchmarkComparison/Program.fs:9]: It would be interesting to test it with different JIT and PGO options in future.

**Review by vzarytovskii — APPROVED** [review]

**Comment by kerams** [comment]
How deterministic are the allocated allocated byte numbers? In theory they should be constant for the same input. I'm asking because I wonder if I could use this to see what kind of impact my changes are having as I'm making them.

**Pull request merged**